ACPs/151-icm-use-current-block-pchain-height/README.md

```diff
@@ -0,0 +1,38 @@
+| ACP           | 151                                                             |
+| :------------ | :-------------------------------------------------------------- |
+| **Title**     | Use current block P-Chain height for ICM signature verification |
+| **Author(s)** | Ian Suvak ([@iansuvak](https://github.com/iansuvak))            |
+| **Status**    | Proposed                                                        |
+| **Track**     | Standards                                                       |
+
+## Abstract
+
+Proposes that VMs use the P-Chain height of the current block being built instead of the parent block when verifying aggregated signatures of Avalanche Interchain Messages (ICM). This will allow for a more reliable way to determine which validators should participate in signing the message, and remove unnecessary waiting periods.
+
+## Motivation
+
+Currently ICM messages use P-Chain height of the parent block when verifying messages for inclusion in the current block. Using the parent's P-Chain height is necessary for safe block building purposes but not for ICM message verification. 
+
+Using the P-Chain height of the current block being built would make operations using ICM messages to modify the validator set, such as ones specified in [ACP-77](../77-reinventing-subnets/README.md) be verifiable sooner and more reliably. Currently at least two new P-Chain blocks need to be produced after the relevant state change for it to be reflected for purposes of ICM aggregate signature verification.
+
+## Specification
+
+[Block Context](https://github.com/ava-labs/avalanchego/blob/d2e9d12ed2a1b6581b8fd414cbfb89a6cfa64551/snow/engine/snowman/block/block_context_vm.go#L14) containing `PChainHeight` field is being passed to the VMs building the inner block, and is later being used by the VMs to fetch the canonical validator set for verification of ICM aggregated signatures. 
+
+The height currently passed in is the P-Chain height of the parent block which is the same height used by the ProposerVM to verify the proposer. This is necessary for proposer verification but not for verifying the state of this block so should be populated with the P-Chain height of the block being currently built instead. 
```

Comment:
```suggestion
The [block context](https://github.com/ava-labs/avalanchego/blob/d2e9d12ed2a1b6581b8fd414cbfb89a6cfa64551/snow/engine/snowman/block/block_context_vm.go#L14) contains a `PChainHeight` field that is passed from the ProposerVM to the inner VMs building the block. It is later used by the inner VMs to fetch the canonical validator set for verification of ICM aggregated signatures.

The `PChainHeight` currently passed in by the ProposerVM is the P-Chain height of the parent block. The proposed change is to instead have the ProposerVM pass in the P-Chain height of the current block.
```

The motivation already covers why this is safe and why it's useful, so think it's more concise to just include the exact specification of the change here.